src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+/**
+ * Parses user command and executes tasks based on user commands.
+ */
+public class Parser {
+    private String userCommand;
+
+    /**
+     * Constructs new <code>Parser</code> object.
+     * @param command the command taken from user input.
+     */
+    public Parser(String command) {
+        this.userCommand = command;
+    }
+
+    /**
+     * Parses and responds to the user command.
+     * @throws DukeException when user command is invalid.
+     */
+    public void parseAndRespond() throws DukeException {
+        String[] splitted = userCommand.split(" ", 2);
+        switch (splitted[0]) {
+        case "list":
+            Ui.listTasks();
+            break;
+        case "mark":
+            if (splitted.length <= 1) {
+                throw new DukeException("Please indicate which task to mark!");
+            } else {
+                int taskNo = Integer.parseInt(splitted[1]);
+                TaskList.mark(taskNo);
+                Storage.save();
+            }
+            break;
+        case "unmark":
+            if (splitted.length <= 1) {
+                throw new DukeException("Please indicate which task to unmark!");
+            } else {
+                int taskNo = Integer.parseInt(splitted[1]);
+                TaskList.unmark(taskNo);
+                Storage.save();
+            }
+            break;
+        case "find":
+            if (splitted.length <= 1) {
+                throw new DukeException("Please provide a term to search for! (⋟﹏⋞)");
+            } else {
+                String searchTerm = splitted[1];
+                TaskList.searchFor(searchTerm);
+            }
+            break;
+        case "todo":
+            if (splitted.length <= 1) {
+                throw new DukeException("Please provide a description for this todo! (⋟﹏⋞)");
+            } else {
+                String todoTask = splitted[1];
+                Todo newTodo = new Todo(todoTask);
+                TaskList.add(newTodo, "todo");
+            }
+            break;
+        case "deadline":
+            if (splitted.length <= 1) {
+                throw new DukeException("Please provide a description for this deadline! (⋟﹏⋞)");
+            } else {
+                String[] deadTask = splitted[1].split("/by");
+                if (deadTask.length == 1) {
```

Comment:
Could be considered deep nesting with affects readability. Maybe consider employing a command class?

Reply:
Good idea, I'll try to create a command class for all the commands and their responses.